src/app/services/endpoints.service.spec.ts

```diff
@@ -3,13 +3,152 @@ import { TestBed, inject } from '@angular/core/testing';
+    expect(endpointsService.contactURL()).toBe('web/contact/');
+  });
+  it('should return user details url' , () => {
+    expect(endpointsService.userDetailsURL()).toBe('auth/user/');
+  });
+  it('should return change password url' , () => {
+    expect(endpointsService.changePasswordURL()).toBe('auth/password/change/');
+  });
+  it('should return verify Email url' , () => {
+    expect(endpointsService.verifyEmailURL()).toBe('auth/registration/verify-email/');
+  });
+  it('should return featured challenges url' , () => {
+    expect(endpointsService.featuredChallengesURL()).toBe('challenges/featured/');
+  });
+  it('should return subscribe url' , () => {
+    expect(endpointsService.subscribeURL()).toBe('web/subscribe/');
+  });
+  it('should return participant team url' , () => {
+    expect(endpointsService.participantTeamURL(2)).toBe('participants/participant_team/2');
+  });
+  it('should return team url' , () => {
+    expect(endpointsService.ourTeamURL()).toBe('web/team/');
+  });
+  it('should return invite participant member url' , () => {
+    expect(endpointsService.participantTeamInviteURL(2)).toBe('participants/participant_team/2/invite');
+  });
+  it('should return all challenges url' , () => {
+    expect(endpointsService.allChallengesURL('past')).toBe('challenges/challenge/past');
+    expect(endpointsService.allChallengesURL('future')).toBe('challenges/challenge/future');
+    expect(endpointsService.allChallengesURL('present')).toBe('challenges/challenge/present');
+  });
+  it('should return all host teams url' , () => {
+    expect(endpointsService.allHostTeamsURL()).toBe('hosts/challenge_host_team');
+  });
+  it('should return invite member to host team url' , () => {
+    expect(endpointsService.hostTeamInviteURL(2)).toBe('hosts/challenge_host_teams/2/invite');
+  });
+  it('should return challenge detail url' , () => {
+    expect(endpointsService.challengeDetailsURL(2)).toBe('challenges/challenge/2/');
+  });
+  it('should return challenge star url' , () => {
+    expect(endpointsService.challengeStarsURL(2)).toBe('challenges/2/');
+  });
+  it('should return challenge participant teams url' , () => {
+    expect(endpointsService.challengeParticipantTeamsURL(2)).toBe('participants/participant_teams/challenges/2/user');
+  });
+  it('should return participate url' , () => {
+    expect(endpointsService.challengeParticipateURL(2, 3)).toBe('challenges/challenge/2/participant_team/3');
+  });
+  it('should return challenge phase url' , () => {
+    expect(endpointsService.challengePhaseURL(2)).toBe('challenges/challenge/2/challenge_phase');
+  });
+  it('should return update challenge detail url' , () => {
+    expect(endpointsService.updateChallengePhaseDetailsURL(2, 3)).toBe('challenges/challenge/2/challenge_phase/3');
+  });
+  it('should return challenge phase split url' , () => {
+    expect(endpointsService.challengePhaseSplitURL(2)).toBe('challenges/2/challenge_phase_split');
```

Comment:
Whenever possible, try not to hardcode URLs. Make them variables at the top and use them.